sway/input/cursor.c

```diff
@@ -1220,6 +1220,9 @@ static void handle_request_set_cursor(struct wl_listener *listener,
 
 void cursor_set_image(struct sway_cursor *cursor, const char *image,
 		struct wl_client *client) {
+	if (!(cursor->seat->wlr_seat->capabilities & WL_SEAT_CAPABILITY_POINTER)) {
```

Comment:
What happens when the seat looses the capability? This prevents the image from being unset.

Reply:
It gets removed in `seat_update_capabilities`.

Reply:
Ah, but we've set the caps already. I'll amend.

Reply:
My bad, `seat_update_capabilities` doesn't call `cursor_set_image` (calls directly `wlr_cursor_set_image`).